app/controllers/admin/blocked_ips_controller.rb

```diff
@@ -0,0 +1,46 @@
+#frozen_string_literal: true
+
+class Admin
+  class BlockedIpsController < ApplicationController
+    before_action :authenticate_user!
+    before_action :admin_required
+
+    layout "application"
+
+    def index
+      @q_blocked_by = params[:q_blocked_by].try( :strip )
+      @q_ip = params[:q_ip].try( :strip )
+
+      @blocked_ips = BlockedIp.joins( :user ).all
+      @blocked_ips = @blocked_ips.where( "users.login LIKE ?", "%#{@q_blocked_by}%" ) unless @q_blocked_by.blank?
+      @blocked_ips = @blocked_ips.where( "blocked_ips.ip LIKE ?", "%#{@q_ip}%" ) unless @q_ip.blank?
+      @blocked_ips = @blocked_ips.select( "blocked_ips.*, users.login as login" )
+      respond_to do | format |
+        format.html { render layout: "admin" }
+      end
+    end
+
+    def block
```

Comment:
There's a few rails model lifecycle practices that might be good to stick to here. This is a BlockedIps controller, which has a index page, and allows the creation and deletion of BlockedIps. The rails custom is use `create` for the create method (with the `POST` method), and `destroy` for the destroy method (instead of the custom `block` and `unblock` used here). Here are some changes we could make to have it act more consistently with other controllers, e.g.:

```diff
diff --git a/app/controllers/admin/blocked_ips_controller.rb b/app/controllers/admin/blocked_ips_controller.rb
index 070e8fa60..d8b6a4593 100644
--- a/app/controllers/admin/blocked_ips_controller.rb
+++ b/app/controllers/admin/blocked_ips_controller.rb
@@ -4,6 +4,7 @@ class Admin
   class BlockedIpsController < ApplicationController
     before_action :authenticate_user!
     before_action :admin_required
+    before_action :load_record, only: [:destroy]

     layout "application"

@@ -20,11 +21,11 @@ class Admin
       end
     end

-    def block
+    def create
       blocked_ip = BlockedIp.find_or_initialize_by( ip: params[:ip] )

       unless blocked_ip.valid?
-        flash[:notice] = "Failed to block IP `#{ip}`: #{blocked_ip.errors.full_messages.to_sentence}"
+        flash[:notice] = "Failed to block IP `#{params[:ip]}`: #{blocked_ip.errors.full_messages.to_sentence}"
         return redirect_back_or_default( action: "index" )
       end

@@ -34,13 +35,18 @@ class Admin
       redirect_to admin_blocked_ips_path( q_ip: params[:q_ip], q_blocked_by: params[:q_blocked_by] )
     end

-    def unblock
-      blocked_ip = BlockedIp.find_by( ip: params[:ip] )
-      blocked_ip&.destroy
+    def destroy
+      @blocked_ip.destroy

       Rails.cache.delete( "blocked_ips" )

       redirect_to admin_blocked_ips_path( q_ip: params[:q_ip], q_blocked_by: params[:q_blocked_by] )
     end
+
+    protected
+
+    def load_record( options = {} )
+      super( options.merge( klass: BlockedIp ) )
+    end
   end
 end
```

* leverage `load_record` to load instances for instance endpoints, like destroy
* customize `load_record` since it doesn't get the instance class correct by default since this controller is nested under Admin
* Update block -> create, and unblock ->  destroy
* Destroy the instance created with `load_record` (a 404 wold be generated if the instance didn't exist)

If you did want to go with this, then the routes would need to be updated as well, e.g.:

```diff
diff --git a/config/routes.rb b/config/routes.rb
index b0aed26c3..f548bf0a0 100644
--- a/config/routes.rb
+++ b/config/routes.rb
@@ -596,11 +596,9 @@ Rails.application.routes.draw do
         get :pending
       end
     end
-    resources :blocked_ips, only: :index, controller: "admin/blocked_ips" do
+    resources :blocked_ips, only: [:index, :create, :destroy], controller: "admin/blocked_ips" do
       collection do
         get :index
-        put :block
-        delete :unblock
       end
     end
   end
```

* remove the custom methods
* specify that we only have the three defined methods, :index, :create, and :destroy

Lastly, some changes would need to be made to the view for the proper paths and method, e.g.:
```diff
diff --git a/app/views/admin/blocked_ips/index.html.haml b/app/views/admin/blocked_ips/index.html.haml
index 878513b1a..0de5d69a5 100644
--- a/app/views/admin/blocked_ips/index.html.haml
+++ b/app/views/admin/blocked_ips/index.html.haml
@@ -24,7 +24,7 @@
                 =link_to 'Reset', url_for, class: "btn btn-link"
     .col-xs-6
       #controls.well
-        =form_tag( { controller: :blocked_ips, action: :block }, method: :put ) do
+        =form_tag( { controller: :blocked_ips }, method: :post ) do
           .row
             .col-xs-3
               .text-right
@@ -56,5 +56,5 @@
                 %td= blocked_ip.login
                 %td= blocked_ip.created_at
                 %td
-                  = button_to unblock_admin_blocked_ips_path( ip: blocked_ip.ip ), method: :delete, class: "btn btn-warning" do
+                  = button_to admin_blocked_ip_path( blocked_ip ), method: :delete, class: "btn btn-warning" do
                     Unblock
```